client/state/selectors/is-site-checklist-complete.js

```diff
@@ -0,0 +1,54 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import getSiteChecklist from 'state/selectors/get-site-checklist';
+import isSiteChecklistLoading from 'state/selectors/is-site-checklist-loading';
+import { getSiteFrontPage } from 'state/sites/selectors';
+
+/**
+ * Checks whether the tasklist has been completed.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @return {String} null if there's no data yet, false if the tasklist is incomplete, true if it's complete.
+ */
+export default function isSiteChecklistComplete( state, siteId ) {
+	const siteChecklist = getSiteChecklist( state, siteId );
+
+	if (
+		isSiteChecklistLoading( state, siteId ) ||
```

Comment:
I had to remove this line to prevent a loop where the data is constantly requested. I think the render kept happening before the network request completed. Swapping to using `hasChecklistData` in the main component would probably do it too.

Reply:
This was confusing me and the problem comes from here I think https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/checklist/main.jsx#L194 The `QuerySiteChecklist` component is rendered regardless of whether the data is available, but then that means that `isSiteChecklistLoading` keeps returning `true` and we get stuck in a loop. 

I feel like there should be a better pattern for retrieving this data 🤔

Reply:
I noticed–even before embedding the Checklist in the CH–that the Checklist constantly sends requests. They're doing that on purpose I guess, maybe to be constantly get results like which task was just completed? In any case, I think it's beyond the scope of this PR.
And I think `isSiteChecklistLoading` does return false briefly. Enough time to render the checklist.